### PR TITLE
Add Windows env to CI

### PR DESCRIPTION
**Pull request by ambreeen**
### What does this PR do?
Adds testing in a Windows environment.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -49,10 +89,26 @@ jobs:
     steps:
       - build-artifacts
 
+  package-extension:
+    description: 'Packages extenson in a Windows environment'
+    executor: win/default
+    environment:
+      - SFDX_AUTOUPDATE_DISABLE: true
+      - SFDX_DOMAIN_RETRY: 300
+      - NODE_VERSION: '12.4.0'
```
Comment (lcampos) [inline .circleci/config.yml]: If you're going to enforce node version `12.4.0` you should update the engine vscode api in package.json from `^1.30.0` to `^1.40.0`. This is because version 1.30.0 of vscode runs on nodejs 8 (https://code.visualstudio.com/updates/v1_30#_continued-electron-30-exploration). They introduced node 12 in version 1.40.0.

If you want to keep the extension usable for vscode versions `1.30.0` and above then you should have your tests running in node 8, node 10 and node 12.

File: .circleci/config.yml [hunk 2]
```diff
@@ -36,6 +36,46 @@ commands:
...
+            nvm install $NODE_VERSION
+            nvm use $NODE_VERSION
+            nvm install-latest-npm
+      - run:
+          name: 'Install dependent modules'
+          shell: bash.exe
+          command: |
+            npm config set scripts-prepend-node-path true
+            npm install
+            npm install -g sfdx-cli@7.45.1
```
Comment (lcampos) [inline .circleci/config.yml]: You should change this to install the latest version of `sfdx-cli`. We had to default it to `7.45.1` because of a performance issue that has been resolved and we'll be changing that in the next days.

File: .circleci/config.yml [hunk 3]
```diff
@@ -36,6 +36,46 @@ commands:
           paths:
             - node_modules
 
+  build-win:
```
Comment (lcampos) [inline .circleci/config.yml]: This is unrelated to these changes. Your extension is simpler than the salesforcedx-vscode and does not have the need of using lerna to build it. You should look into using yarn instead of npm since it's faster, provides better lock file functionality, among other things.

Comment (ambreeen) [reply .circleci/config.yml]: I'll look into using yarn instead of npm for a future release. Thanks!

File: .circleci/config.yml (hunk 1 above)
Comment (ambreeen) [reply .circleci/config.yml]: Thanks for the feedback. I updated the vscode version in `package.json`

File: package.json [hunk 4]
```diff
@@ -137,6 +137,7 @@
     "vscode:prepublish": "cd client && npm run update-vscode && cd .. && npm run compile",
     "compile": "tsc -b",
     "watch": "tsc -b -w",
-    "postinstall": "cd client && npm install && cd .."
+    "postinstall": "cd client && npm install && cd ..",
```
Comment (lcampos) [inline package.json]: I didn't noticed this before, you could remove your use of `cd` by using npm's `--prefix` flag and this script would look something like `npm install --prefix client`
